pkg/smart/common.go

```diff
@@ -59,23 +59,79 @@ func detectSCSIType(name string) (Dev, error) {
 	return &device, nil
 }
 
-// readDeviceCapacity sends a SCSI READ CAPACITY(16) command to a device and
-// returns the capacity in bytes.
+// sendReadCap10 is used to send a SCSIReadCapacity10 scsi command to
+// get logical block address and logical size of a disk
+func (d *SCSIDev) sendReadCap10() (uint32, uint32, error) {
+	respBuf10 := make([]byte, 8) // respBuf for read capacity(10) scsi cmd
+
+	// Populate the CDB10 required fields to send SCSIReadCapacity10 command
+	cdb10 := CDB16{SCSIReadCapacity10}
```

Comment:
Should this be CDB10?

Reply:
Right @kmova, it should be CDB10.
It will work with CDB16 also but better to keep CDB10 here.